src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -150,12 +152,116 @@ begin
   simp [e₂, direct_sum.submodule_coe, direct_sum.to_module, dfinsupp.sum_add_hom_apply]
 end
 
-/-- An orthonormal basis on a fintype `ι` for an inner product space induces an isometry with
+end
+
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.single [decidable_eq ι] (i : ι) (a : 𝕜) :
+  euclidean_space 𝕜 ι :=
+pi.single i a
+
+@[simp] theorem euclidean_space.single_apply [decidable_eq ι] (i : ι) (a : 𝕜) (j : ι) :
+  (euclidean_space.single i a) j = ite (j = i) a 0 :=
+begin
+  rw [euclidean_space.single],
+  rw ← pi.single_apply i a j,
+end
+
+lemma euclidean_space.inner_single_left [decidable_eq ι] (i : ι) (a : 𝕜) (v : euclidean_space 𝕜 ι) :
+  ⟪euclidean_space.single i (a : 𝕜), v⟫ = conj a * (v i) :=
+by simp [apply_ite conj]
+
+lemma euclidean_space.inner_single_right [decidable_eq ι] (i : ι) (a : 𝕜)
+  (v : euclidean_space 𝕜 ι) :
+  ⟪v, euclidean_space.single i (a : 𝕜)⟫ =  a * conj (v i) :=
+by simp [apply_ite conj, mul_comm]
```

Comment:
I might consider swapping these to the order naturally produced,  `conj (v i) * a`, so there is no `mul_comm`.  No strong opinion though, do as you prefer.